### PR TITLE
v1.12 backports 2022-09-07

### PR DESCRIPTION
**Pull request by joamaki**
* #20112 -- Fix conflicting routes for multiple ENIs in IPAM mode (@recollir)
 * #21080 -- pkg/k8s: fix node update sync from k8s to kvstore (@aanm)
 * #21039 -- ~~Add workload name and kind into L7 flows (@chancez)~~
 * #21093 -- pkg/k8s/watcher: fix deadlock with service event handler & CES watcher. (@tommyp1ckles)
 * #20795 -- agent: Stop serving DNS before shutdown (@nebril)
 * #21115 -- EgressGW: make logging less verbose (@julianwiedmann)
 * #21131 -- metallb: bump to latest metallb version (@ldelossa)
 * #20896 -- if the user do not specify a proxy port and it fails to listen on the port, instead of retrying again for that specific one, increment it. (@NikhilSharmaWe)
 * #21133 -- operator: do not GC kvstore nodes if CiliumNodes are not available (@aanm)
 * #21146 -- datapath: tolerate missing ifaces when setting rp_filter sysctl (@julianwiedmann)
 * #21126 -- datapath: allow local NodePort traffic for `eni+` container interfaces with CNI chaining (@ti-mo)
 * #20819 -- pkg/bgpv1/annotations: Optimize annotations Errors (@MikeLing)
 * #20848 -- Coalesce of health endpoint CIDRs (@dezmodue)
 * #21127 -- hubble-ui: release v0.9.2 (@geakstr)
 * #21051 -- install: Support configuring metricsRelabelings on ServiceMonitors (@chancez)
 * #21202 -- operator: update CiliumNode in kvstore without lease (@tklauser)
 * #21135 -- filter out pod labels from synchronizing with cilium endpoint labels (@NikhilSharmaWe)
 * #21124 -- Use pod Deployment name as workload name for flow workload field (@chancez)
 * #21220 -- k8s: fix test flake in TestGenerateToCIDRFromEndpoint. (@tommyp1ckles)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 20112 21080 21093 20795 21115 21131 20896 21133 21146 21126 20819 20848 21127 21051 21202 21135 21124 21220; do contrib/backporting/set-labels.py $pr done 1.12; done
```
or with
```
$ make add-label BRANCH=v1.12 ISSUES=20112,21080,21093,20795,21115,21131,20896,21133,21146,21126,20819,20848,21127,21051,21202,21135,21124,21220
```

### CODE REVIEW

**Review by geakstr — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Review by tklauser — APPROVED** [review]
Backport of my change looks good, thanks.

**Review by julianwiedmann — APPROVED** [review]

**Review by chancez — APPROVED** [review]
My patches LGTM

**Review by ldelossa — APPROVED** [review]
My changes look good, thanks.

**Review by tommyp1ckles — APPROVED** [review]

**Comment by tommyp1ckles** [comment]
Added a note on the other backport PR relating to a flake fix: 

https://github.com/cilium/cilium/pull/21223#issuecomment-1239992718

Is it possible to add it to this set as well?

File: pkg/option/config.go [hunk 1]
```diff
@@ -2855,6 +2861,7 @@ func (c *DaemonConfig) Populate() {
 	c.KeepConfig = viper.GetBool(KeepConfig)
 	c.KVStore = viper.GetString(KVStore)
 	c.KVstoreLeaseTTL = viper.GetDuration(KVstoreLeaseTTL)
+	c.CNIChainingMode = viper.GetString(CNIChainingMode)
```
Comment (ti-mo) [inline pkg/option/config.go]: Nit: the original commit had this sitting between `ClusterMeshConfig` and `DatapathMode`.

**Review by ti-mo — COMMENTED** [review]
One nit that could maybe potentially confuse future backports.

**Comment by aanm** [comment]
@joamaki can we drop https://github.com/cilium/cilium/pull/21039 from the backport. We discussed with @chancez and we will not be backporting this PR, at least not for now.

**Comment by nebril** [comment]
/test-backport-1.12

**Review by ti-mo — APPROVED** [review]
Thanks!

**Comment by nebril** [comment]
GKE and 1.21 tests failed in BeforeAll phase, retriggering.

**Comment by nebril** [comment]
/test-1.21-4.9

**Comment by nebril** [comment]
`k8s-1.24-kernel-net-next` failed because of a flake in K8sNode tests

**Pull request merged**